### PR TITLE
Remove calls to `AnnotationRegistry::registerLoader()`

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | no
| Deprecations? | yes
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

Calling `AnnotationRegistry::registerLoader()` should not be necessary anymore since Annotations 1.10 (April 2020), as long as all annotations are autoloadable. Let's remove all calls to that method.

### CODE REVIEW

File: src/Symfony/Bridge/PhpUnit/composer.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
         "symfony/error-handler": "For tracking deprecated interfaces usages at runtime with DebugClassLoader"
     },
     "conflict": {
+        "doctrine/annotations": "<1.10",
```
Comment (stof) [inline src/Symfony/Bridge/PhpUnit/composer.json:31]: this conflict should also be added in FrameworkBundle, as it is the one configuring the annotation reader service

**Comment by derrabus** [comment]
~~Technically, we could do this on 5.4 already because `AnnotationRegistry::registerLoader('class_exists');` should be a no-op and we already have conflict rules for Annotations < 1.10 in place. Let me know if you think I should backport the change.~~

The low-deps tests showed me that we should not backport. Registering `class_exists` as loader is still necessary, if someone somewhere calls (the also deprecated) `AnnotationRegistry::registerFile()`. Removing our only call to that method required me to bump the ORM to >= 2.12.

We cannot anticipate if a codebase still has that `AnnotationRegistry::registerFile()` call somewhere, so we might as well decide to not merge this change as long as we support Annotations v1.

File: src/Symfony/Bridge/PhpUnit/composer.json (hunk 1 above)
Comment (derrabus) [reply src/Symfony/Bridge/PhpUnit/composer.json:31]: We already have a conflict rule in place there:

https://github.com/symfony/symfony/blob/b23a087d7b4a9b84a1dde314ee9a900dadd741db/src/Symfony/Bundle/FrameworkBundle/composer.json#L73

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/annotations.php [hunk 2]
```diff
@@ -23,13 +23,10 @@
...
         ->set('annotations.reader', AnnotationReader::class)
-            ->call('addGlobalIgnoredName', [
-                'required',
-                service('annotations.dummy_registry')->ignoreOnInvalid(), // dummy arg to register class_exists as annotation loader only when required
-            ])
+            ->call('addGlobalIgnoredName', ['required'])
 
         ->set('annotations.dummy_registry', AnnotationRegistry::class)
-            ->call('registerUniqueLoader', ['class_exists'])
+            ->deprecate('symfony/framework-bundle', '6.3', 'The "%service_id%" service is deprecated and will be removed without replacement.')
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/annotations.php]: I'd suggest just dropping this, nobody can use it today

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/annotations.php (hunk 2 above)
Comment (derrabus) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/annotations.php]: Dropped.

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

**Comment by nicolas-grekas** [comment]
I reverted the part about phpunit-bridge in 9f5c00567537d2b6ba7265a9c74af85aecd44329

**Comment by derrabus** [comment]
Fair enough, but the ORM bump in the root composer.json was still legit. Why did you revert it as well?

**Comment by nicolas-grekas** [comment]
By mistake, let me fix that.
Fixed in 6d3b3f161b

**Comment by derrabus** [comment]
Thanks!

**Pull request merged**